main.go

```diff
@@ -92,6 +94,37 @@ func exportArtifacts(artifacts []gradle.Artifact, deployDir string) ([]string, e
 	return paths, nil
 }
 
+func filterUtilityVariants(variants []string) []string {
+	ignoredSuffixes := [...]string{"Classes", "Resources", "UnitTestClasses", "AndroidTestClasses", "AndroidTestResources"}
+	var filteredVariants []string
+
+	for _, v := range variants {
+		shouldIgnore := false
+		for _, suffix := range ignoredSuffixes {
+			if strings.HasSuffix(v, suffix) {
+				shouldIgnore = true
+				break
+			}
+		}
+
+		if !shouldIgnore {
+			filteredVariants = append(filteredVariants, v)
+		}
+	}
+
+	return filteredVariants
+}
+
+func separateVariants(variantsAsOneLine string) []string {
+	variants := strings.Split(variantsAsOneLine, newLine)
+
+	for _, variant := range variants {
+		variant = strings.TrimSpace(variant)
```

Comment:
This deos not actually assign a new value in the slice :(. You need to use variants[i] = instead.